applications/RANSApplication/custom_utilities/rans_variable_utilities.cpp

```diff
@@ -230,6 +234,239 @@ void CopyNodalSolutionStepVariablesList(
 
     KRATOS_CATCH("");
 }
+
+template <typename TDataType>
+void AssignConditionVariableValuesToNodes(ModelPart& rModelPart,
+                                          const Variable<TDataType>& rVariable,
+                                          const Flags& rFlag,
+                                          const bool FlagValue)
+{
+    auto& r_nodes = rModelPart.Nodes();
+    VariableUtils().SetHistoricalVariableToZero(rVariable, r_nodes);
+
+    const int number_of_conditions = rModelPart.NumberOfConditions();
+#pragma omp parallel for
+    for (int i_cond = 0; i_cond < number_of_conditions; ++i_cond)
+    {
+        auto& r_cond = *(rModelPart.ConditionsBegin() + i_cond);
+        if (r_cond.Is(rFlag) == FlagValue)
+        {
+            const int number_of_nodes = r_cond.GetGeometry().PointsNumber();
+            const auto& r_normal = r_cond.GetValue(rVariable);
+            for (int i_node = 0; i_node < number_of_nodes; ++i_node)
+            {
+                auto& r_node = r_cond.GetGeometry()[i_node];
+                r_node.SetLock();
+                r_node.FastGetSolutionStepValue(rVariable) +=
+                    r_normal * (1.0 / static_cast<double>(number_of_nodes));
+                r_node.UnSetLock();
+            }
+        }
+    }
+
+    rModelPart.GetCommunicator().AssembleCurrentData(rVariable);
+}
+
+void AddAnalysisStep(ModelPart& rModelPart, const std::string& rStepName)
```

Comment:
```suggestion
void AddAnalysisStep(
    ModelPart& rModelPart,
    const std::string& rStepName)
```

(also the ones below)